src/lib/common/sol-platform-impl-linux-micro.c

```diff
@@ -846,16 +846,10 @@ sol_platform_impl_get_serial_number(char **number)
     return r;
 }
 
-char *
-sol_platform_impl_get_os_version(void)
+int
+sol_platform_impl_get_os_version(char **version)
```

Comment:
you can avoid this by moving `sol_platform_impl_get_os_version()` to `sol-platform-linux-common.c`, then it's automatically shared in both systemd and linux-micro and you don't need to replicate it.